File: backend/app/models/project_type.py
"""Admin-managed project types."""

from __future__ import annotations

from datetime import datetime
from typing import TYPE_CHECKING, Optional

from sqlalchemy import Boolean, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base

if TYPE_CHECKING:
    from app.models.project import ProjectDetails


class ProjectType(Base):
    __tablename__ = "project_types"

    id: Mapped[int] = mapped_column(primary_key=True)
    code: Mapped[str] = mapped_column(String(30), unique=True, nullable=False, index=True)
    name_en: Mapped[str] = mapped_column(String(100), nullable=False)
    name_bn: Mapped[Optional[str]] = mapped_column(String(150), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default="true")
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, server_default="0")
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )

    projects: Mapped[list["ProjectDetails"]] = relationship(back_populates="project_type")

File: frontend/src/types/projectType.ts
export interface ProjectType {
  id: number;
  code: string;
  name_en: string;
  name_bn?: string | null;
  is_active: boolean;
  sort_order: number;
  created_at: string;
}

export interface ProjectTypeCreatePayload {
  code: string;
  name_en: string;
  name_bn?: string;
  is_active?: boolean;
  sort_order?: number;
}

export interface ProjectTypeUpdatePayload {
  code?: string;
  name_en?: string;
  name_bn?: string;
  is_active?: boolean;
  sort_order?: number;
}
